python/cuml/cuml/ensemble/randomforest_common.pyx

```diff
@@ -313,6 +313,15 @@ class BaseRandomForestModel(UniversalBase):
         self.dtype = np.float64
         self.update_labels = False
         super().cpu_to_gpu()
+        # Set fitted attributes not transferred by treelite, but only when the
+        # accelerator is active.
+        # We only transfer "simple" attributes, not np.ndarrays or DecisionTree
+        # instances, as these could be used by the GPU model to make predictions.
+        # Which means they need to be of the type that the GPU model expects.
+        if GlobalSettings().accelerator_active:
+            for name, value in vars(self._cpu_model).items():
```

Comment:
I don't love copying over any scalars in the model `__dict__` willy-nilly - is there any way we could do this with a known list of fitted attributes instead?

Reply:
There is unfortunately no explicit way to inspect an estimator get the list of fitted attributes. But, all scikit-learn estimators follow the convention of "All and only fitted attributes end in a `_`." The one foot gun is to check that the attribute name doesn't also start with a `_` to avoid private attributes.

I've not checked but I am 99% sure that this is enforced as part of the common estimator tests.

Reply:
I think Jim is proposing to actually hard-code a set of expected parameter names that we are specifically trying to address with this work-around.

Reply:
We can hard code the list of attributes or detect them based on their name. Either works for me

Reply:
I think hard-coding would be the way to go.